### PR TITLE
server: add HTTP API to enable/disable async commit and 1PC

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Problem Summary:

After upgrading a TiDB cluster, tools like tiup may automatically enable async commit in a cluster. 
Tools cannot set global variables directly, so we need to provide an API to do it.

Related issue: https://github.com/tikv/sig-transaction/issues/75

### What is changed and how it works?
What's Changed:

Two HTTP APIs were added:

```shell
curl -X POST -d "tidb_enable_async_commit=1" http://{TiDBIP}:10080/settings
curl -X POST -d "tidb_enable_async_commit=0" http://{TiDBIP}:10080/settings
```

```shell
curl -X POST -d "tidb_enable_1pc=1" http://{TiDBIP}:10080/settings
curl -X POST -d "tidb_enable_1pc=1" http://{TiDBIP}:10080/settings
```

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: server/http_handler.go [hunk 1]
```diff
@@ -720,6 +721,46 @@ func (h settingsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				return
 			}
 		}
+		if asyncCommit := req.Form.Get("tidb_enable_async_commit"); asyncCommit != "" {
+			s, err := session.CreateSession(h.Store.(kv.Storage))
```
Comment (jackysp) [inline server/http_handler.go:725]: I think you need to close the session somewhere.

File: docs/tidb_http_api.md [hunk 2]
```diff
@@ -477,3 +477,18 @@ timezone.*
     * op=nowait: return after binlog status is recoverd, do not wait until the skipped-binlog transactions are committed.
     * op=reset: reset `SkippedCommitterCounter` to 0 to avoid the problem that `SkippedCommitterCounter` is not cleared due to some unusual cases.
     * op=status: Get the current status of binlog recovery.
+
+1. Enable/disable async commit feature
```
Comment (jackysp) [inline docs/tidb_http_api.md:481]: This may cause a security hazard. Previously, this kind of APIs affected at most one tidb-server.

Comment (sticnarf) [reply docs/tidb_http_api.md:481]: What security risk can there be if we provide such an API with very restricted function?

Comment (jackysp) [reply docs/tidb_http_api.md:481]: It bypasses the privilege check. I don't think it's necessary to do this for me. If it is really necessary, some risk is acceptable.

Comment (jackysp) [reply docs/tidb_http_api.md:481]: OK, I know the reason.

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by youjiali1995 — APPROVED** [review]
/LGTM

**Pull request merged**